### DISCUSSION TITLE
Enable Devtools only in development mode?

### DISCUSSION THREAD

**Discussion by lukemorales** [TanStack/query#2381] [Q&A] [answered]
Is there a reason why Devtools is only disabled on production mode, instead of enabling it only on development mode?

What brings me this question, is that my team had an issue with it being a part of our tests snapshots and generating a bunch of warnings, so we had to verify it manually to display on `development` mode only. I know that taking it away from the test wrapper is also another solution, but was it intentional to have Devtools being disabled only on production mode? Would it be ok for me to open a PR adjusting it to display only in development or this was an intended behavior @tannerlinsley @TkDodo?

**Comment by tannerlinsley** [comment, marked as answer]
`mode === development` or `mode !== production` mostly mean the same thing to me and most users of the library, but I can see how `mode = testing` would present issues with the latter. I would entertain this PR for sure.

**Reply by lukemorales** [reply]
Yeah, I see the point you're talking about! But since the source code is just verifying if `NODE_ENV === 'production'` that got me suspicious if enabling it in all other envs, including `testing` was intentional. Gonna open a PR adjusting this then! Thanks @tannerlinsley
